salt/utils/win_functions.py

```diff
@@ -121,9 +121,16 @@ def get_current_user():
     '''
     try:
         user_name = win32api.GetUserNameEx(win32api.NameSamCompatible)
-        if user_name[-1] == '$' and win32api.GetUserName() == 'SYSTEM':
+        if user_name[-1] == '$':
             # Make the system account easier to identify.
-            user_name = 'SYSTEM'
+            # Fetch sid so as to handle other language than english
+            test_user = win32api.GetUserName()
+            sid, d, t = win32security.LookupAccountName('', test_user)
+            real_sid = user = win32security.ConvertSidToStringSid(sid)
+            if test_user == 'SYSTEM':
+                user_name = 'SYSTEM'
+            elif real_sid == 'S-1-5-18':
+                user_name = 'SYSTEM'
     except pywintypes.error as exc:
```

Comment:
Just a style comment. To make this slightly more efficient (eg avoiding `win32security.LookupAccountName()` when unnecessary), I would change the structure to be something like:

```
test_user = win32api.GetUserName()
if test_user == 'SYSTEM':
    user_name = 'SYSTEM'
else:
    sid = win32security.LookupAccountName(None, test_user)[0]
    sid_str = win32security.ConvertSidToStringSid(sid)
    if sid_str == 'S-1-5-18':
        user_name = 'SYSTEM'
```

I also borrowed a little bit of the style from the already existing function `get_sid_from_name()`. Here are some changes due to following the style of the preexisting function: `None` instead of `''`, `[0]` to only get the first element from the return tuple, not using an extra `user` variable for `win32security.ConvertSidToStringSid(sid)` which doesn't seem like it is referenced.